File: Data Structures & Algorithms/top-k-elements-in-list/submission-9.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
        
        Dictionary<int,int> freq=new Dictionary<int,int>();

        foreach (int ch in nums){

            if (freq.ContainsKey(ch)){
                freq[ch]+=1;
            }
            else{
                  freq[ch]=1;
            }
        }

        List<List<int>> buckets=new List<List<int>>();

        for (int i=0; i<nums.Length+1; i++){
            buckets.Add(new List<int>());
        }

        foreach(var item in freq){
            buckets[item.Value].Add(item.Key);
        }

        List<int> res=new List<int>();

        for (int i=buckets.Count-1; i>=0; i--){

            foreach(int bucket in buckets[i]){
                res.Add(bucket);

                if (res.Count==k){
                    return res.ToArray();
                }
            }
        }
        return res.ToArray();

    }
}

File: Data Structures & Algorithms/minimum-window-with-characters/submission-9.py
class Solution:
    def minWindow(self, s: str, t: str) -> str:
        
        min_len=float("inf")

        freq={}

        for ch in t:
            if ch in freq:
                freq[ch]+=1
            else:
                freq[ch]=1


        count=len(freq)

        i=0
        j=0

        while j<len(s):

            if s[j] in freq:
                freq[s[j]]-=1

                if freq[s[j]]==0:
                    count-=1

            if count==0:

                while count==0:

                    if s[i] in freq:
                        freq[s[i]]+=1

                        if freq[s[i]]==1:
                            count+=1

                            if j-i+1<min_len:
                                min_len=j-i+1
                                start=i

                    i+=1

            j+=1

        return ("" if min_len==float("inf") else s[start:start+min_len])









